spec/p2p/v0.34/switch.md

```diff
@@ -0,0 +1,239 @@
+accepted by the transport configured for the switch.
+
+The `Accept` method of the `Transport` returns a `Peer` with which an inbound
+connection was established.
+The maximum number of inbound peers, i.e., peers from which a connection was
+accepted is determined by the `MaxNumInboundPeers` configuration parameter,
+whose default value is `40`.
+The switch accepts the peer if the maximum number of inbound peers was not
+reached, or if the peer was configured as an _unconditional peer_.
+If accepted, the peer is added to the switch using the `addPeer` method.
+
+If the switch does not accept the established incoming connection, or if the
+`addPeer` method returns an error, the switch invokes the `Transport.Cleanup`
+method to clean any resources associated with the peer.
+
+> TODO: cleaning up a peer should mean, in practice, disconnecting from it.
+
+The transport's `Accept` method can also return a number of errors.
+Errors of `ErrRejected` or `ErrFilterTimeout` types are ignored,
+an `ErrTransportClosed` causes the accepted routine to be interrupted,
+while other errors cause the routine to panic.
+
+> TODO: ErrRejected has a multitude of causes.
+> TODO: which source of errors can cause the routine to panic?
+
+## addPeer
+
+This method adds a peer to the switch,
+either after dialing (by `addOutboundPeerWithConfig`, called by `DialPeerWithAddress`)
+a peer and establishing an outbound connection,
+or after accepting (`acceptRoutine`) a peer and establishing an inbound connection.
+
+The first step is to invoke the `filterPeer` method.
+It checks whether the peer is already in the set of connected peers,
+and whether any of the configured `peerFilter` methods reject the peer.
+If the peer is already present or it is rejected by any filter, the `addPeer`
+method fails and returns an error.
+
+Then, the new peer is started, added to the set of connected peers, and added
+to all reactors
+More precisely, first the new peer's information is first provided to every
+reactor (`InitPeer` method).
+Next, the peer's sending and receiving routines are started, and the peer is
+added to set of connected peers.
+These two operations can fail, causing `addPeer` to return an error.
+Then, in the absence of previous errors, the peer is added to every reactor (`AdPeer` method).
+
+> Adding the peer to the peer set returns a `ErrSwitchDuplicatePeerID` error
+> when a peer with the same ID is already presented.
+>
+> TODO: Starting a peer could be reduced as starting the MConn with that peer?
+
+## Broadcast
+
+This method broadcasts a message on a channel, by sending the message in
+parallel to all connected peers.
+
+The method spawns a thread for each connected peer, invoking the `Send` method
+provided by each `Peer` instance with the provided message and channel ID.
+The return value (a boolean) of these calls are redirected to a channel that is
+returned by the method.
+
+> TODO: does anyone use this method? Check.
```

Comment:
I see it used in:
- consensus `broadcastNewRoundStepMessage`, ` broadcastNewValidBlockMessage` , 1broadcastHasVoteMessage`; 
- statesync (within `Sync`)

Reply:
Ok, so it is in use.

Reply:
Maintained as a TODO, but mentioning these uses.